exasol/shared_memory_vault.py

```diff
@@ -0,0 +1,226 @@
+from typing import Optional
```

Comment:
Would using a [keyring](https://pypi.org/project/keyring/) also solve your problem?

Reply:
I think the main motivation for developing an in-memory vault was to make sure the password gets destroyed when the docker container with DSS stops. Or ideally even earlier, when the last notebook is closed. You are probably right - the implementation could just as well be based on keyring.

Reply:
I already looked at keyring stuff, and it seems to cause more issues then solving.